[aoti] Use generate_fake_kernels_from_real_mismatches config for draft exported programs

yushangdi opened the pull request:
Summary:
Sometimes we get `MetadataMismatchError` in aoti compilation because draft export uses the flag below to infer the fake kernel when there’s a mismatch, but aoti doesn’t have this flag turned on.

https://fburl.com/code/9qzytl6q
 torch._functorch.config.generate_fake_kernels_from_real_mismatches


If we set this flag to True, then aoti compilation would work.

Test Plan:
```
buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r aoti_runtime_asserts
```

Differential Revision: D72345085




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

angelayi commented on test/inductor/test_aot_inductor.py:
```diff
@@ -3401,23 +3400,21 @@ def forward(self, a, b):
-            # optimized = torch._inductor.aoti_load_package(pt2_file)
+            with functorch_config.patch(
+                {"generate_fake_kernels_from_real_mismatches": need_config_patch}
+            ):
```
can you write a TODO  that this should no longer be needed after https://github.com/pytorch/pytorch/pull/150093

yushangdi committed:
[aoti] Use generate_fake_kernels_from_real_mismatches config for draft exported programs (#150651)

Summary:

Sometimes we get `MetadataMismatchError` in aoti compilation because draft export uses the flag below to infer the fake kernel when there’s a mismatch, but aoti doesn’t have this flag turned on.

https://fburl.com/code/9qzytl6q
 torch._functorch.config.generate_fake_kernels_from_real_mismatches


If we set this flag to True, then aoti compilation would work.

Test Plan:
```
buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r aoti_runtime_asserts
```

Reviewed By: muchulee8

Differential Revision: D72345085
